### PR TITLE
[DRAFT] Run tests in codemod file with `bowler test`

### PR DESCRIPTION
**Pull request by marionlb**
Addresses #65 

`bowler run path/to/codemod.py` runs all the tests in codemod.py

### CODE REVIEW

File: bowler/main.py [hunk 1]
```diff
@@ -138,5 +138,24 @@ def run(codemod: str, argv: List[str]) -> None:
...
+@click.argument("codemod", required=True, type=str)
+def test(codemod: str) -> None:
+    """
+    Run the tests in the codemod file 
+        """
+    import unittest
+    import importlib.util
+    import os.path
+
+    module_name_from_codemod = os.path.basename(codemod).replace(".py", "")
```
Comment (thatch) [inline bowler/main.py]: I would expect this import code to be pretty similar to the 'run' case above, can you check?

File: bowler/main.py [hunk 2]
```diff
@@ -138,5 +138,24 @@ def run(codemod: str, argv: List[str]) -> None:
...
+    spec = importlib.util.spec_from_file_location(module_name_from_codemod, codemod)
+    foo = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(foo)
+    suite = unittest.TestLoader().loadTestsFromModule(foo)
+
+    unittest.TextTestRunner().run(suite)
```
Comment (thatch) [inline bowler/main.py]: Can you make sure this exits nonzero on failure?  Looking at what unittest.main does is probably useful here.

File: bowler/main.py [hunk 3]
```diff
@@ -138,5 +138,24 @@ def run(codemod: str, argv: List[str]) -> None:
         sys.argv[1:] = original_argv
 
 
+@main.command()
+@click.argument("codemod", required=True, type=str)
+def test(codemod: str) -> None:
+    """
+    Run the tests in the codemod file 
+        """
+    import unittest
```
Comment (thatch) [inline bowler/main.py]: These imports should be fine to include at module-level.

File: bowler/main.py [hunk 4]
```diff
@@ -138,5 +138,24 @@ def run(codemod: str, argv: List[str]) -> None:
         sys.argv[1:] = original_argv
 
 
+@main.command()
+@click.argument("codemod", required=True, type=str)
+def test(codemod: str) -> None:
+    """
+    Run the tests in the codemod file 
+        """
```
Comment (thatch) [inline bowler/main.py]: Can you dedent this line please?

**Comment by thatch** [comment]
@marionlb are you able to continue on this? If not i will add the test and get it merged for the next release

**Comment by thatch** [comment]
Now includes a test.  @jreese thoughts on whether it should take args like unittest.main (e.g. `-v`)

**Comment by jreese** [comment]
Taking `-v` would be useful, but not necessary. Your call.

**Comment by thatch** [comment]
Let's take it as-is.  I would personally use `python -m unittest`, but this is a verb I looked for early on.

**Pull request merged**